mlflow/langchain/__init__.py

```diff
@@ -127,6 +128,41 @@ def get_default_conda_env():
     return _mlflow_conda_env(additional_pip_deps=get_default_pip_requirements())
 
 
+def _get_databricks_auth_env_vars():
+    """
+    To initialize WorkspaceClient correctly in a subprocess in databricks, we need to pass
+    some environment variables to the subprocess. This function tries to fetch the required
+    environment variables locally and raises an exception if they are not found.
+
+    Note:
+        Currently we only support PAT auth and OAuth service principal.
+    """
+    envs = {}
+    from databricks.sdk import WorkspaceClient
+
+    w = WorkspaceClient()
+    auth_type = w.config.auth_type
+    if "DATABRICKS_HOST" not in os.environ:
+        envs["DATABRICKS_HOST"] = w.config.host
+    if auth_type == "pat":
+        if "DATABRICKS_TOKEN" not in os.environ:
+            envs["DATABRICKS_TOKEN"] = w.config.token
+    elif auth_type == "oauth-m2m":
+        if "DATABRICKS_CLIENT_ID" not in os.environ:
+            envs["DATABRICKS_CLIENT_ID"] = w.config.client_id
+        if "DATABRICKS_CLIENT_SECRET" not in os.environ:
+            envs["DATABRICKS_CLIENT_SECRET"] = w.config.client_secret
+    else:
+        raise MlflowException(
+            "Currently we only support PAT auth and OAuth service principal "
+            "for UC functions execution in Databricks. Please use one of them "
+            "or open an issue on GitHub if you need other auth types."
```

Comment:
```suggestion
            "or open an issue on GitHub for unsupported auth types."
```

Reply:
Just realized databricks auth env vars are added https://github.com/mlflow/mlflow/blob/10c3833d2347eed8ae14c2defe2b62d85a18008b/mlflow/utils/requirements_utils.py#L335-L336 Updated the code!